src/test/java/uk/gov/cshr/service/security/LoginAttemptServiceTest.java

```diff
@@ -1,79 +1,82 @@
-    public void loginSucceedSetAttemptsToZero() {
-        String email = "test@domain.com";
+    private IdentityService identityService;
 
-        loginAttemptCache.put(email, 3);
+    @Mock
+    private IdentityRepository identityRepository;
 
-        loginAttemptService.loginSucceeded(email);
+    private LoginAttemptService loginAttemptService;
 
-        assertEquals(new Integer(0), loginAttemptCache.get(email));
+    @Before
+    public void setUp(){
+        identityService = new IdentityService("", identityRepository, null, null, null, null);
+        loginAttemptService = new LoginAttemptService(maxLoginAttempts, identityService);
     }
 
     @Test
-    public void loginFailedIncrementsFailedAttempts() {
+    public void loginSucceedSetAttemptsToZero() {
         String email = "test@domain.com";
+        Identity identity = new Identity();
+        identity.setFailedLoginAttempts(3L);
 
-        loginAttemptCache.clear();
-
-        when(identityService.existsByEmail(email)).thenReturn(true);
+        when(identityRepository.findFirstByEmailEquals(email)).thenReturn(identity);
 
-        loginAttemptService.loginFailed(email);
+        loginAttemptService.loginSucceeded(email);
 
-        assertEquals(new Integer(1), loginAttemptCache.get(email));
 
-        verify(identityService, times(0)).lockIdentity(email);
+        assertEquals(new Long(0), identityService.getFailedLoginAttempts(email));
     }
 
-    @Test
-    public void loginFailedDoesNotIncrementCacheIfIdentityDoesNotExist() {
-        String email = "test@domain.com";
-
-        loginAttemptCache.clear();
+   @Test
+    public void loginFailedIncrementsFailedAttempts() {
+       String email = "test2@domain.com";
+       Identity identity = new Identity();
+       identity.setFailedLoginAttempts(0L);
 
-        when(identityService.existsByEmail(email)).thenReturn(false);
+       when(identityRepository.findFirstByEmailEquals(email)).thenReturn(identity);
+       when(identityRepository.existsByEmail(email)).thenReturn(true);
 
-        loginAttemptService.loginFailed(email);
+       loginAttemptService.loginFailed(email);
 
-        assertNull(loginAttemptCache.get(email));
+       assertEquals(new Long(1), identityService.getFailedLoginAttempts(email));
     }
 
 
     @Test
     public void loginFailedLocksIdentityWhenMaxAttemptsIsExceeded() {
         String email = "test@domain.com";
+        Identity identity = new Identity();
+        identity.setFailedLoginAttempts(3L);
 
-        loginAttemptCache.put(email, 3);
-
-        when(identityService.existsByEmail(email)).thenReturn(true);
+        when(identityRepository.findFirstByActiveTrueAndEmailEquals(email)).thenReturn(identity);
+        when(identityRepository.findFirstByEmailEquals(email)).thenReturn(identity);
+        when(identityRepository.existsByEmail(email)).thenReturn(true);
 
         try {
             loginAttemptService.loginFailed(email);
         } catch (AuthenticationException e) {
```

Comment:
can we add a new statement after line 74
`fail("AuthenticationException was not thrown")`

Reply:
will add